### PR TITLE
[Impeller] allow for one more active swapchain image.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Currently the rendering performance is less stable with this patch: https://flutter-flutter-perf.skia.org/e/?queries=device_type%3DPixel_7_Pro%26sub_result%3D90th_percentile_frame_rasterizer_time_millis%26sub_result%3D99th_percentile_frame_rasterizer_time_millis%26sub_result%3Daverage_frame_rasterizer_time_millis%26sub_result%3Dworst_frame_rasterizer_time_millis%26test%3Dcomplex_layout_scroll_perf_impeller__timeline_summary&selected=commit%3D40592%26name%3D%252Carch%253Dintel%252Cbranch%253Dmaster%252Cconfig%253Ddefault%252Cdevice_type%253DPixel_7_Pro%252Cdevice_version%253Dnone%252Chost_type%253Dlinux%252Csub_result%253D99th_percentile_frame_rasterizer_time_millis%252Ctest%253Dcomplex_layout_scroll_perf_impeller__timeline_summary%252C

The reason I suspect that is that we're signaling the semaphore at the start of the raster thread. This means that we're limiting the actual number of pending presents to 1, as we also need to account for the CPU access. Increase this and the max texture count to 3, to allow 2 pending presents and one image worked on by the CPU.

### CODE REVIEW

File: impeller/renderer/backend/vulkan/swapchain/ahb/ahb_swapchain_impl_vk.cc [hunk 1]
```diff
@@ -17,9 +17,11 @@ namespace impeller {
 
 //------------------------------------------------------------------------------
 /// The maximum number of presents pending in the compositor after which the
-/// acquire calls will block.
+/// acquire calls will block. This value is 2 images given to the system
```
Comment (chinmaygarde) [inline impeller/renderer/backend/vulkan/swapchain/ahb/ahb_swapchain_impl_vk.cc:20]: The value is not 2 though.

File: impeller/renderer/backend/vulkan/swapchain/ahb/ahb_texture_pool_vk.h [hunk 2]
```diff
@@ -45,7 +45,7 @@ class AHBTexturePoolVK {
   explicit AHBTexturePoolVK(
       std::weak_ptr<Context> context,
       android::HardwareBufferDescriptor desc,
-      size_t max_entries = 2u,
+      size_t max_entries = 3u,
```
Comment (chinmaygarde) [inline impeller/renderer/backend/vulkan/swapchain/ahb/ahb_texture_pool_vk.h:48]: This count is the number of textures idling and used neither by the client (us) or the compositor. I don't think holding onto an additional texture matters here.

**Review by chinmaygarde — APPROVED** [review]
Just the `kMaxPendingPresents` should get rid of the wait. See the comment on discord for why this was added. It only handles a runaway situation on the emulators where there were too many pending presents without a completion callback. Ideally we should never see a wait. That we are means we can bump up this number.

File: impeller/renderer/backend/vulkan/swapchain/ahb/ahb_swapchain_impl_vk.cc (hunk 1 above)
Comment (jonahwilliams) [reply impeller/renderer/backend/vulkan/swapchain/ahb/ahb_swapchain_impl_vk.cc:20]: 2+1 = 3

File: impeller/renderer/backend/vulkan/swapchain/ahb/ahb_texture_pool_vk.h (hunk 2 above)
Comment (jonahwilliams) [reply impeller/renderer/backend/vulkan/swapchain/ahb/ahb_texture_pool_vk.h]: Removed. I think I must be seeing the case where the texture ages quicker than expected.

**Comment by gaaclarke** [comment]
Here's the drop for this PR.  It's about a 15% reduction in `average_frame_rasterizer_time_millis:platform_views_scroll_perf_impeller__timeline_summary`.  Combined with the swapchain chain it's about a 35% reduction.

https://flutter-flutter-perf.skia.org/e/?keys=X01917a97c6e2c484ca1b758071aff6e6&selected=commit%3D40608%26name%3D%252Carch%253Dintel%252Cbranch%253Dmaster%252Cconfig%253Ddefault%252Cdevice_type%253DSM-A025V%252Cdevice_version%253Dnone%252Chost_type%253Dlinux%252Csub_result%253Daverage_frame_rasterizer_time_millis%252Ctest%253Dplatform_views_scroll_perf_impeller__timeline_summary%252C&xbaroffset=40591

**Pull request merged**